test/src/unittests/tonal/test_pitchsaliencefunctionpeaks.py

```diff
@@ -0,0 +1,174 @@
+
+
+from numpy import *
+from essentia_test import *
+
+class TestPitchSalienceFunctionPeaks(TestCase):
+  
+    def testInvalidParam(self):
+        self.assertConfigureFails(PitchSalienceFunctionPeaks(), {'binResolution': -1})
+        self.assertConfigureFails(PitchSalienceFunctionPeaks(), {'maxFrequency': -1})
+        self.assertConfigureFails(PitchSalienceFunctionPeaks(), {'minFrequency': -1})
+        self.assertConfigureFails(PitchSalienceFunctionPeaks(), {'referenceFrequency': -1})                
+
+    def testEmpty(self): 
+        self.assertRaises(RuntimeError, lambda: PitchSalienceFunctionPeaks()([]))
+    
+    # Check for empty output arrays for a salience function with all zero values
+    def testZeros(self): 
+        bins, values = PitchSalienceFunctionPeaks()(zeros(128))
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(values, [])    
+
+    # Check for empty output arrays for a salience function repeated values of different lengths
+    def testConstantInputs(self): 
+        bins, values = PitchSalienceFunctionPeaks()([0.5, 0.5, 0.5])
+        self.assertEqualVector(bins, [])    
+        self.assertEqualVector(values, [])    
+        bins, values = PitchSalienceFunctionPeaks()(ones(128))     
+        self.assertEqualVector(bins, [])    
+        self.assertEqualVector(values, [])    
+
+    # Provide a single input peak with a unit magnitude at the reference frequency, and 
+    # validate that the output salience function has only one non-zero element at the first bin.
+    def testSinglePeak(self):
+        freq_speaks = [55] # length 1
+        mag_speaks = [1] # length 4
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)    
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [0.])
+        self.assertEqualVector(values, [1.])   
+        #self.assertEqual(calculatedPitchSalience[0],1)
+
+    # As above but with Harmonic Weight = 0
+    def testSinglePeakHw0(self):
+        freq_speaks = [55] # length 1
+        mag_speaks = [1] # length 4
+        calculatedPitchSalience = PitchSalienceFunction(harmonicWeight=0)(freq_speaks,mag_speaks)    
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [0.])
+        self.assertEqualVector(values, [1.])   
+        #self.assertEqual(calculatedPitchSalience[0],1)
+    
+    # As above but with Harmonic Weight = 1
+    def testSinglePeakHw1(self):
+        freq_speaks = [55] # length 1
+        mag_speaks = [1] # length 4
+        calculatedPitchSalience = PitchSalienceFunction(harmonicWeight=1)(freq_speaks,mag_speaks)    
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [0.])
+        self.assertEqualVector(values, [1.])   
+        #self.assertEqual(calculatedPitchSalience[0],1)        
```

Comment:
There's no harmonics, only a single frequency peak as an input, therefore testing both harmonicWeight = 0 and 1 is redundant.